ts/Data/Formula/FormulaParser.ts

```diff
@@ -0,0 +1,448 @@
+/* *
+ *
+ *  (c) 2009-2023 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Sophie Bremer
+ *
+ * */
+
+
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+
+import type {
+    Arguments,
+    Formula,
+    Function,
+    Operator,
+    Range,
+    Reference,
+    Value
+} from './FormulaTypes.js';
+
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+
+/**
+ * Formula parser might struggle over a syntax error.
+ * @private
+ */
+export interface FormulaParserError extends Error {
+    message: string;
+    name: 'FormulaParseError';
+}
+
+
+/* *
+ *
+ *  Constants
+ *
+ * */
+
+
+/**
+ * @private
+ */
+const booleanRegExp = /^(?:FALSE|TRUE)/;
+
+
+/**
+ * @private
+ */
+const decimal1RegExp = /^[+-]?\d+(?:\.\d+)?(?:e[+-]\d+)?/;
+
+
+/**
+ * @private
+ */
+const decimal2RegExp = /^[+-]?\d+(?:,\d+)?(?:e[+-]\d+)?/;
```

Comment:
What is the difference between `decimal2RegExp` and `decimal1RegExp` ? Add comment please.